### PR TITLE
feat: helm chart

### PR DESCRIPTION
**Pull request by tchataigner**
This PR refines our k8s configuration and overall experience around our docker images. I could properly create a cluster and deploy the Helm Chart to it with the latest [`ethereum-proof-server`](https://github.com/argumentcomputer/zk-light-clients/pkgs/container/ethereum-proof-server/274147815?tag=latest) image.

## Changelog

- Create an [`eksctl`](https://eksctl.io/) configuration file to easily deploy a cluster on AWS with the machines needed for our proof servers. 
- Refactor k8s configuration to an Helm Chart
- Refactor the Helm Chart values and the k8s values to make sure that a client can properly interact with a proof server and generate proofs
- Refactor the `Dockerfile` for our proof server so that the `target-cpu` and `target` work for our targeted server specifications (`x86-64-v4`)

### CODE REVIEW

File: docker/proof-server-chart/values.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      containerPort: "8080"
+      rustLog: "debug"
+      rustMinStack: "16777216"
+      shardSize: "4194304"
+      shardBatchSize: "0"
+      shardChunkingMultiplier: "32"
+      reconstructCommitments: "false"
+    image:
+      repository: ghcr.io/argumentcomputer/ethereum-proof-server
+      tag: latest
```
Comment (huitseeker) [inline docker/proof-server-chart/values.yaml:17]: Our tag uses latest yet we set an `appVersion` of 0.1.0, does this make sense?

File: docker/proof-server-chart/templates/proof-server-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        readinessProbe:
+          httpGet:
+            path: /ready
+            port: 8080
+          initialDelaySeconds: 15
+          periodSeconds: 10
+        name: proof-server
+        ports:
+        - containerPort: 8080
+        resources: {}
```
Comment (huitseeker) [inline docker/proof-server-chart/templates/proof-server-deployment.yaml:74]: The resources field being empty makes me wonder: how would things work if we wanted to migrate out of the nodeSelector of EKS?

Would a minimum usable amount of hardware be useful here? @samuelburnham may have ideas.

```
resources:
  requests:
    memory: "128Gi"
    cpu: "16"
  limits:
    memory: "128Gi"
    cpu: "32"
```

File: docker/proof-server-chart/templates/proof-server-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        - name: CONTAINER_PORT
+          value: {{ quote .Values.deployment.proofServer.env.containerPort }}
+        - name: KUBERNETES_CLUSTER_DOMAIN
+          value: {{ quote .Values.kubernetesClusterDomain }}
+        image: {{ .Values.deployment.proofServer.image.repository }}:{{ .Values.deployment.proofServer.image.tag
+          | default .Chart.AppVersion }}
+        imagePullPolicy: {{ .Values.deployment.proofServer.imagePullPolicy }}
+        livenessProbe:
+          httpGet:
+            path: /health
```
Comment (huitseeker) [inline docker/proof-server-chart/templates/proof-server-deployment.yaml]: Do we want to put the path of the health/readiness probes configurable via `values.yaml`?

File: docker/proof-server-chart/templates/proof-server-hpa.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+spec:
+  maxReplicas: 5
+  metrics:
+  - resource:
+      name: cpu
+      target:
+        averageUtilization: 50
+        type: Utilization
+    type: Resource
+  minReplicas: 2
```
Comment (huitseeker) [inline docker/proof-server-chart/templates/proof-server-hpa.yaml]: We may want to make min/max Replicas, and averageUtilization configurable via values.yaml.

File: docker/proof-server-chart/values.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+deployment:
```
Comment (huitseeker) [inline docker/proof-server-chart/values.yaml:1]: This globally looks great, but we may want to document the non-obivous / magic values in this file.

File: docker/proof-server-chart/values.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  proofServer:
+    env:
+      containerPort: "8080"
+      rustLog: "debug"
+      rustMinStack: "16777216"
```
Comment (huitseeker) [inline docker/proof-server-chart/values.yaml:8]: Here we may want to document that we have that same value in the Docker image too.

**Review by huitseeker — COMMENTED** [review]
Globally, this looks good, but I've left inline comments.

File: docker/README.md [hunk 7]
```diff
@@ -12,48 +12,104 @@ To ease the execution and deployment of our Light Client implementation, we prov
...
+them as defined in the project documentation.
 
-- `APTOS_NODE_URL`: The URL of the Aptos node (e.g., `127.0.0.1:8080`).
+### Helm Chart
 
-**Ethereum variables**
+To deploy the proof servers and the client using Kubernetes, we provide a Helm chart in the `proof-server-chart`
+directory. The pre-requisite to run the Helm chart is to have a Kubernetes cluster running.
+To help get one running we have a dedicated `eksctl` configuration file in the `k8s` directory.
+It leverages [`AWS EKS`](https://aws.amazon.com/eks/) to create a cluster taht we can leverage  to deploy the Helm chart.
```
Comment (samuelburnham) [inline docker/README.md]:
```suggestion
It leverages [`AWS EKS`](https://aws.amazon.com/eks/) to create a cluster that we can leverage to deploy the Helm chart.
```

File: docker/proof-server-chart/templates/proof-server-deployment.yaml (hunk 2 above)
Comment (samuelburnham) [reply docker/proof-server-chart/templates/proof-server-deployment.yaml:74]: Yes this seems like a good idea. The `r7iz.metal-32xl` instance has 128 vCPUs and 1024GB RAM, which would be a good limit value, but I'm not sure what the right `request` value would be if not equivalent to `limits`. cc @tchataigner

File: docker/proof-server-chart/values.yaml (hunk 1 above)
Comment (tchataigner) [reply docker/proof-server-chart/values.yaml:17]: I changed `appVersion` to latest. However, as It is specified in the documentation that the user should replace the image by what they want it is likely that those will never have a 1:1 correspondence (if the user does not bother to maintain it).

File: docker/proof-server-chart/templates/proof-server-deployment.yaml (hunk 2 above)
Comment (tchataigner) [reply docker/proof-server-chart/templates/proof-server-deployment.yaml:74]: I think that we want to use `r7iz.metal-32xl ` anyway as we base our benchmark on these. If some users want another type of hardware they can always change the configuration. 

To be 100% honest I'm not even sure how to properly test this workflow outside of an `eksctl create` then `helm install` so making this change might take some time to validate.

I created #234 for it.

**Review by huitseeker — APPROVED** [review]
This looks great, thanks a lot!!

**Pull request merged**